akita/src/queryEntity.ts

```diff
@@ -129,12 +130,11 @@ export class QueryEntity<S extends EntityState, EntityType = getEntityType<S>, I
   selectMany<R>(ids: IDType[], project?: (entity: EntityType) => R): Observable<EntityType[] | R[]> {
     if (!ids || !ids.length) return of([]);
 
-    const entities = ids.map(id => this.selectEntity(id, project));
-
-    return combineLatest(entities).pipe(
-      map(v => v.filter(Boolean)),
-      auditTime(0)
-    );
+    return this.select(state => state.entities).pipe(
+      map(entities => ids.map(id => getEntity(id, project)(entities))),
+      map((entities): R[] | EntityType[] => entities.filter(Boolean)),
```

Comment:
Thanks!

1. I know that it was there before, but we can save the second redundant iteration, and filter the `undefined` entities in the first loop.
2. Did you check that the typings are still working as expected?

Reply:
1. I created a new helper method that uses a reducer to be able to skip entities without a second loop.
2. Yes, the typings are still correct. `getEntity` is the same as before, and `distinctUntilArrayItemChanged` will get either an array of type _R_ or _EntityType_, and will also return an array of the same type.